sdk/resourcemanager/Azure.ResourceManager/src/Resources/Custom/PolicyAssignmentData.cs

```diff
@@ -57,8 +57,42 @@ internal PolicyAssignmentData(ResourceIdentifier id, string name, ResourceType r
 
         /// <summary> The location of the policy assignment. Only required when utilizing managed identity. </summary>
         public AzureLocation? Location { get; set; }
+
+#pragma warning disable CS0618 // This type is obsolete and will be removed in a future release.
+        private SystemAssignedServiceIdentity _identity;
         /// <summary> The managed identity associated with the policy assignment. </summary>
-        public SystemAssignedServiceIdentity Identity { get; set; }
+        [Obsolete("This property is obsolete and will be removed in a future release. Please use ManagedIdentity.", false)]
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public SystemAssignedServiceIdentity Identity
+        {
+            get => _identity;
+            set
+            {
+                _identity = value;
+                _managedIdentity = value == null ? null : _identity.Identity;
+            }
+        }
+
+        private ManagedServiceIdentity _managedIdentity;
+        /// <summary> The managed identity associated with the policy assignment. </summary>
+        public ManagedServiceIdentity ManagedIdentity
+        {
+            get => _managedIdentity;
+            set
+            {
+                _managedIdentity = value;
+                if (value == null)
+                    _identity = null;
+                else
+                {
+                    if (_identity == null)
```

Comment:
can't we just have line 90?  Since we just need to change the internal memory the reference of SystemAssignedServiceIdentity doesn't need to change here.

Reply:
In this codnition branch when we set a non-null value to ManagedIdentity, _identity could initially be null, therefore we need to create a new instance for it if it's null before we can update its internal memory of ManagedServiceIdentity.